test/sql/hnsw_create_expr.sql

```diff
@@ -13,9 +13,56 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
 
+CREATE OR REPLACE FUNCTION int_to_dynamic_binary_real_array(n INT) RETURNS REAL[] AS $$
+DECLARE
+    binary_string TEXT;
+    real_array REAL[] := '{}';
+    i INT;
+    result_length INT;
+BEGIN
+    binary_string := lpad(CAST(n::BIT(3) AS TEXT), 3, '0');
```

Comment:
what is the number `3` doing in here?
can you write a comment clarifying what this and the next functions do?

Reply:
Added comments there